### ISSUE TITLE
How to make pdf-tools buffer refresh right after corresponding pdf file changes?

### ISSUE THREAD

**Issue by ustcscgy** [politza/pdf-tools#25]
First of all, thanks for making such a great tool!

My question is, the buffer updates its content a few seconds after the pdf file changes,  is there a variable can make the update immediately?

This problem emerges when I try to use pdf-tools as the pdf viewer of WhizzyTeX, which is an emacs minor mode for real-time incremental viewing of LaTeX documents. WhizzyTeX updates pdf file constantly, so I wish pdf-tools can refresh as soon as possible.

**Comment by politza** [comment]
`(setq auto-revert-interval 0.5)` ?

The display updates when the buffer is reverted.

**Comment by ustcscgy** [comment]
Yes, you are right. Why didn't I think of that?

Also run `(auto-revert-set-timer)` to make sure the new value take effect.

Thanks.

**Comment by ragloo** [comment]
@ustcscgy  
Please could you share your configuration and advice how to use pdf-tools with WhizzyTeX.

I'm very interested in your feedback.
Many thanks :)

**Comment by ustcscgy** [comment]
@ragloo 

Set whizzytex to use the pdf engine (instead of dvi), then just use whizzytex minor-mode on the tex buffer, then use emacs to open the generated pdf file (with pdf-tools, of course, and with `auto-revert` settings).

So you can have two buffers side by side, one is tex buffer, one is pdf preview buffer which auto updates
when tex buffer is changed.

**Comment by ragloo** [comment]
@ustcscgy
Many thanks for your feedback, it's very nice from you :)

**Comment by fbob** [comment]
Please could you share your config :)
I've tried to setup whizzytex to work with pdf-tools without success.
It works with advi but for pdf the process end without any output and this error message "Mode switched off externally"

**Comment by Nidish96** [comment]
Hey, following some of the above guidelines I did manage to set up auto-revert with pdf-tools (yay!), but there's a small snag that's irritating me.
When I set it up with doc-view, the update is very seamless. But of course, doc-view is not searchable and has issues and I don't want to use it.
![docview](https://user-images.githubusercontent.com/17080531/45777219-ced46e00-bc1a-11e8-9805-83a0d58a4abe.gif)
^gif1: Doc-view with latexmk running in the background. Looks good and smooth.

But then when I do the same with pdf-tools, the pdf buffer blinks for a moment before reloading the file. For small files (like in the gif), this is not too long, but for larger documents this really takes some time. Has anyone faced this before/found a way to solve it?
![pdf-tools](https://user-images.githubusercontent.com/17080531/45777291-0511ed80-bc1b-11e8-9a32-075531838816.gif)
^gif2: pdf-tools with latexmk running in the background. The right side buffer blinks each time I save the tex file.
